Framework/Crystal/test/SCDCalibratePanels2Test.h

```diff
@@ -0,0 +1,877 @@
+#include "MantidAPI/Sample.h"
+#include "MantidCrystal/SCDCalibratePanels2.h"
+#include "MantidDataObjects/EventWorkspace.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidGeometry/Crystal/CrystalStructure.h"
+#include "MantidKernel/Logger.h"
+#include "MantidKernel/Unit.h"
+
+#include <boost/filesystem.hpp>
+#include <boost/math/constants/constants.hpp>
+#include <boost/math/special_functions/round.hpp>
+#include <cxxtest/TestSuite.h>
+#include <stdexcept>
+
+using namespace Mantid::API;
+using namespace Mantid::Crystal;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+namespace {
+/// static logger
+Logger g_log("SCDCalibratePanels2Test");
+} // namespace
+
+class SCDCalibratePanels2Test : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static SCDCalibratePanels2Test *createSuite() {
+    return new SCDCalibratePanels2Test();
+  }
+  static void destroySuite(SCDCalibratePanels2Test *suite) { delete suite; }
+
+  // ----------------- //
+  // ----- Setup ----- //
+  // ----------------- //
+  /**
+   * @brief Construct a new SCDCalibratePanels2Test object
+   *
+   */
+  SCDCalibratePanels2Test()
+      : wsname("wsSCDCalibratePanels2Test"),
+        pwsname("pwsSCDCalibratePanels2Test"),
+        tmppwsname("tmppwsSCDCalibratePanels2Test"), // fixed workspace name
+        bank_xtop("bank73/sixteenpack"),             //
+        bank_xcenter("bank12/sixteenpack"),          //
+        bank_xbottom("bank11/sixteenpack"),          //
+        bank_yright("bank59/sixteenpack"),           //
+        bank_yleft("bank58/sixteenpack"),            //
+        bank_ytop("bank88/sixteenpack"),             //
+        bank_ybottom("bank26/sixteenpack"),          //
+        silicon_a(5.431), silicon_b(5.431), silicon_c(5.431),   // angstrom
+        silicon_alpha(90), silicon_beta(90), silicon_gamma(90), // degree
+        silicon_cs(CrystalStructure("5.431 5.431 5.431", "F d -3 m",
+                                    "Si 0 0 0 1.0 0.02")),
+        dspacing_min(1.0), dspacing_max(10.0),   //
+        wavelength_min(0.1), wavelength_max(10), //
+        omega_step(3.0),                         //
+        TOLERANCE_L(5e-4), // this calibration has intrinsic accuracy limit of
+                           // 1mm for translation
+        TOLERANCE_R(1e-5), // this calibration has intrinsic accuracy limit of
+                           // 0.1 deg for rotation
+        LOGCHILDALG(false) {
+    // NOTE:
+    //  The MAGIC PIECE, basically we need to let AlgorithmFactory
+    //  to load a non-related algorithm, then somehow AlgorithmFactory
+    //  can find the Fit algorithm for the remaining test
+    std::shared_ptr<Algorithm> darkmagic =
+        AlgorithmFactory::Instance().create("LoadIsawPeaks", 1);
+    darkmagic->initialize();
```

Comment:
Based on reading examples from other unit tests on Github, both `initialize` and `execute` need to be present.
Since I am not familiar with the function factory, I went with the safer route.

Reply:
dark magic indeed